spec/syncSpec.js

```diff
@@ -641,11 +663,13 @@ function stopSync(dataset) {
  *
  * @param {string} expectedEvent - The name of the event to wait for.
  */
-function waitForSyncEvent(expectedEvent) {
+function waitForSyncEvent(expectedEvent, validator) {
   return function() {
+    const eventValidator = validator || noop;
+
     return new Promise(function(resolve) {
       $fh.sync.notify(function(event) {
-        if (event.code === expectedEvent) {
+        if (event.code === expectedEvent && eventValidator(event)) {
```

Comment:
I think the naming of `eventValidator` is a bit wrong. With the respond immediately changes I noticed that sometimes there'd be multiple notifications when there would usually only have been one. 

This is caused because, when responding immediately, functions like doCreate are not guaranteed to result in a `remote_update_applied` straight away, this could happen on the next loop. So sometimes when we're expecting just one `remote_update_applied` there's a chance of getting two, so this just allows a function that will filter the event further (making sure it has the correct hash etc.).

So it should be called `eventFilter`.